### ISSUE TITLE
Override colors not working when using CoreAnimation

### ISSUE THREAD

**Issue by gustafnilklint** [airbnb/lottie-ios#2150]
When transitioning from Lottie 3.5 -> Lottie 4.2
I noticed that our code to override colors in Lottie animations.

If I go in and configure the AnimationView to use MainThread rendering, the color override is working as expected

The code:

```
let keypath = AnimationKeypath(keys: ["**", "Color"])
animationView.setValueProvider(ColorValueProvider(for: color), keypath: keypath)

```
## Which Version of Lottie are you using?

Lottie 4.2

## Expected Behavior 
The overridden color should appear

## Actual Behavior
The original color is displayed

## Animation JSON
Any animation should do

**Comment by gustafnilklint** [comment]
![Simulator Screen Recording - iPhone 14 Pro - 2023-08-24 at 11 06 48](https://github.com/airbnb/lottie-ios/assets/980447/b55fe2cc-728b-4321-9019-d7c5261ebb40)

The zip-file contains a sample app (from the screen capture above) with the animation file in the project
[LottieSample.zip](https://github.com/airbnb/lottie-ios/files/12427764/LottieSample.zip)

**Comment by calda** [comment]
Thanks for the report, here's a fix: #2154

**Comment by gustafnilklint** [comment]
@calda Thank you for quick diagnose and fix 🙌
May I ask when this will be released? When is your next release planned?
